### PR TITLE
add ensemble_runner.RegionalInput.from_model_fitter and use it for counties

### PR DESCRIPTION
**Pull request by TomGoBravo**
This PR is part of https://trello.com/c/nYgDEl3M/385-ability-to-query-raw-data-by-msa

* In `ensemble_runner` and `webui_data_adaptor_v1` add constructor `Regionalnput.from_model_fitter` which copies data directly from a `ModelFitter` object, saving a round trip to disk.
* Remove `webui_data_adaptor_v1` `generate_state` which was only used for regions in one place.
* Remove `if __main__` in `webui_data_adaptor_v1` that looked like some kind of early test
* Remove `model_fitter.RegionalInput.load_inference_result` which wasn't used from other classes
* Remove `execute_model_for_fips` which added a branch that always went one way
* Move the `states_only=False` part of `model_fitter.run_state` to it's own function because it had a single caller and complicated using `Regionalnput.from_model_fitter`
* Adds use of a tempdir to `test_pyseir_end_to_end_idaho` so it doesn't pass when output from a previous run is still in `output/`
* Removes first of two writes of state data, at the end of `_state_only_pipeline` and inline the remaining caller of `cli._map_outputs`

### CODE REVIEW

File: pyseir/cli.py [hunk 1]
```diff
@@ -50,42 +49,16 @@ def _generate_whitelist():
...
-        state_input = webui_data_adaptor_v1.RegionalInput.from_region(state)
-        web_ui_mapper.generate_state(
-            state_input, whitelisted_county_fips=[], states_only=states_only
-        )
-
-
 def _state_only_pipeline(
     region: pipeline.Region, run_mode=DEFAULT_RUN_MODE, output_interval_days=1, output_dir=None,
-):
+) -> model_fitter.ModelFitter:
```
Comment (ghop02) [inline pyseir/cli.py:54]: Can this return the infer rt, and ensembles results as well?

File: pyseir/deployment/webui_data_adaptor_v1.py [hunk 2]
```diff
@@ -68,7 +72,9 @@ def load_inference_result(self) -> Mapping[str, Any]:
         : dict
             Dictionary of fit result information.
         """
-        return load_inference_result(self.region)
+        if self._mle_fit_result is not None:
+            return self._mle_fit_result
+        return pipeline.load_inference_result(self.region)
```
Comment (ghop02) [inline pyseir/deployment/webui_data_adaptor_v1.py:77]: will we ever expect this to be reached?

File: pyseir/cli.py (hunk 1 above)
Comment (ghop02) [reply pyseir/cli.py:54]: It seems like those outputs could easily be passed into the WebUIInputRegion as well. but fine tackling that later

File: pyseir/inference/model_fitter.py [hunk 3]
```diff
@@ -736,52 +718,47 @@ def build_county_list(state: str) -> List[str]:
...
-
-            # Serialize the model results.
-            for fips, fitter in zip(all_fips, fitters):
-                if fitter:
-                    with open(get_run_artifact_path(fips, RunArtifact.MLE_FIT_MODEL), "wb") as f:
-                        pickle.dump(fitter.mle_model, f)
+    return model_fitter
+
+
+def run_counties_of_state(region):
```
Comment (ghop02) [inline pyseir/inference/model_fitter.py:745]: types + a docstring would be handy here!

File: test/pyseir_end_to_end_test.py [hunk 4]
```diff
@@ -12,19 +15,22 @@
...
-    path = pathlib.Path(path)
-    assert path.exists()
-    output = CANPyseirLocationOutput.load_from_path(path)
-    data = output.data
-    with_values = data[schema.RT_INDICATOR].dropna()
-    assert len(with_values) > 10
-    assert (with_values > 0).all()
-    assert (with_values < 6).all()
+
+    with temppathlib.TemporaryDirectory() as tmp_dir:
```
Comment (ghop02) [inline test/pyseir_end_to_end_test.py:19]: I do find the [tmp_path](https://docs.pytest.org/en/stable/tmpdir.html) fixture handy. I know we probably have slightly different styles here :) I think I tend to think that the multiple indentation is more confusing than having pytest take care of it

File: test/pyseir_end_to_end_test.py [hunk 5]
```diff
@@ -12,19 +15,22 @@
...
+        with unittest.mock.patch("pyseir.utils.OUTPUT_DIR", tmp_dir.path):
```
Comment (ghop02) [inline test/pyseir_end_to_end_test.py:20]: This would be handy to make an environment variable I think as well

File: pyseir/cli.py [hunk 6]
```diff
@@ -187,14 +152,8 @@ def _build_all_for_states(
...
-    for state in states:
-        region = pipeline.Region.from_state(state)
-        state_input = webui_data_adaptor_v1.RegionalInput.from_region(region)
-        web_ui_mapper.generate_state(
-            state_input,
-            whitelisted_county_fips=[k for k, v in all_county_fips.items() if v == state],
-            states_only=False,
-        )
+    for fitter in itertools.chain(state_fitters, county_fitters):
+        web_ui_mapper.map_fips(webui_data_adaptor_v1.RegionalInput.from_model_fitter(fitter))
```
Comment (ghop02) [inline pyseir/cli.py:156]: might be nice to rename map_fips without the map - it makes me thing we're running over multiple fips

Comment (ghop02) [inline pyseir/cli.py:156]: This doesn't look like it's in parallel.  we probably want to run this in a pool right?

File: pyseir/cli.py (hunk 1 above)
Comment (TomGoBravo) [reply pyseir/cli.py:54]: Yes, my thought was get this merged with one output, then make another pass for the others.

File: pyseir/deployment/webui_data_adaptor_v1.py (hunk 2 above)
Comment (TomGoBravo) [reply pyseir/deployment/webui_data_adaptor_v1.py:77]: Hopefully not long term. For now it is reached by code that uses the RegionalInput.from_fips constructor, which is called by the non build-all flows.

File: pyseir/inference/model_fitter.py (hunk 3 above)
Comment (TomGoBravo) [reply pyseir/inference/model_fitter.py:745]: Done

File: test/pyseir_end_to_end_test.py (hunk 4 above)
Comment (TomGoBravo) [reply test/pyseir_end_to_end_test.py:19]: I'm torn between my dislike of the fixture magic driven by parameter names and having shorter lines of code :-P

File: test/pyseir_end_to_end_test.py (hunk 5 above)
Comment (TomGoBravo) [reply test/pyseir_end_to_end_test.py:20]: Perhaps, though I haven't been motivated for that so far...

File: pyseir/cli.py (hunk 6 above)
Comment (TomGoBravo) [reply pyseir/cli.py:156]: Done, `write_region`

Comment (TomGoBravo) [reply pyseir/cli.py:156]: Done

**Pull request merged**